provider/mesh/peer.go

```diff
@@ -45,170 +45,9 @@ func openReplace(filename string) (*replaceFile, error) {
-		return err
-	}
-	return f.Close()
-}
-
-// Stop signals the background processing to terminate.
-func (ni *NotificationInfos) Stop() {
-	close(ni.stopc)
-	if err := ni.snapshot(); err != nil {
-		ni.logger.With("err", err).Errorf("Snapshotting failed")
-	}
-}
-
-// Gossip implements the mesh.Gossiper interface.
-func (ni *NotificationInfos) Gossip() mesh.GossipData {
-	return ni.st.copy()
-}
-
-// OnGossip implements the mosh.Gossiper interface.
-func (ni *NotificationInfos) OnGossip(b []byte) (mesh.GossipData, error) {
-	set, err := decodeNotificationSet(b)
-	if err != nil {
-		return nil, err
-	}
-	d := ni.st.mergeDelta(set)
-	// The delta is newly created and we are the only one holding it so far.
-	// Thus, we can access without locking.
-	if len(d.set) == 0 {
-		return nil, nil // per OnGossip contract
-	}
-	return d, nil
-}
-
-// OnGossipBroadcast implements the mesh.Gossiper interface.
-func (ni *NotificationInfos) OnGossipBroadcast(_ mesh.PeerName, b []byte) (mesh.GossipData, error) {
-	set, err := decodeNotificationSet(b)
-	if err != nil {
-		return nil, err
-	}
-	return ni.st.mergeDelta(set), nil
-}
-
-// OnGossipUnicast implements the mesh.Gossiper interface.
-func (ni *NotificationInfos) OnGossipUnicast(_ mesh.PeerName, b []byte) error {
-	set, err := decodeNotificationSet(b)
-	if err != nil {
-		return err
-	}
-	ni.st.mergeComplete(set)
-	return nil
-}
-
-// Set the provided notification information.
-// The expiration timestamp is set to the timestamp plus the configured retention time.
-func (ni *NotificationInfos) Set(ns ...*types.NotificationInfo) error {
-	set := map[notificationKey]notificationEntry{}
-	for _, n := range ns {
-		k := notificationKey{
-			Receiver: n.Receiver,
-			Alert:    n.Alert,
-		}
-		set[k] = notificationEntry{
-			Resolved:  n.Resolved,
-			Timestamp: n.Timestamp,
-			// The expiration timestamp is set at creation time
-			// to avoid synchronization artifacts in garbage collection.
-			ExpiresAt: n.Timestamp.Add(ni.retention),
-		}
-	}
-	update := &notificationState{set: set}
-
-	ni.st.Merge(update)
-	ni.send.GossipBroadcast(update)
-	return nil
-}
-
-// Get the notification information for the given receiver about alerts
-// with the given fingerprints. Returns a slice in order of the input fingerprints.
-// Result entries are nil if nothing was found.
-func (ni *NotificationInfos) Get(recv string, fps ...model.Fingerprint) ([]*types.NotificationInfo, error) {
-	var (
-		res = make([]*types.NotificationInfo, 0, len(fps))
-		k   = notificationKey{Receiver: recv}
-		now = ni.st.now()
-	)
-	for _, fp := range fps {
-		k.Alert = fp
-		if e, ok := ni.st.set[k]; ok && e.ExpiresAt.After(now) {
-			res = append(res, &types.NotificationInfo{
-				Alert:     fp,
-				Receiver:  recv,
-				Resolved:  e.Resolved,
-				Timestamp: e.Timestamp,
-			})
-		} else {
-			res = append(res, nil)
-		}
-	}
-	return res, nil
-}
-
-// Silences provides gossiped silences.
 type Silences struct {
```

Comment:
Yep. Fix will be immediately superseded by #481, which removes the section below.